### PR TITLE
Add setup.py

### PR DESCRIPTION
**Pull request by marianamarasoiu**
This is broadly how setup.py should look for this project.

Before it can be imported with pipenv, we will need to replace the Pipfile-s in the other outstanding PRs with an __init_.py file in each folder and move their dependencies in setup.py. We will also need to create a release to match the version number in setup.py.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from setuptools import setup, find_packages
+
+setup(
+    name="pipelineinfrastructure",
```
Comment (lukechurch) [inline setup.py]: What's the standard naming convention here? Is it 

`pipelineinfrastructure` or `pipeline-infrastructure`

**Comment by lukechurch** [comment]
Thanks very much @marianamarasoiu - I know these few lines cost a lot of python manipulating.

When you say create a release to match the version in setup.py - do you mean a GitHub release? Does it need a specific branch name? Can we reference Github SHAs until then?

Thanks again!

File: setup.py (hunk 1 above)
Comment (marianamarasoiu) [reply setup.py]: As far as I can tell, there's little naming convention on this. Some people use CamelCase, others lower case, either in a single word, or separated by - or _. I used lower case as most python packages that I've used so far seem to be lower case, but I see that CoreDataModules uses upper case as the name, and lowercase for the package. Do you have any preference?

**Comment by marianamarasoiu** [comment]
Yes, I mean a GitHub release, which can be set at a specific commit. I've played with them a bit and seem fairly straightforward to do, and Alexander has been using them for CoreDataModules also from SHAs.

**Pull request merged**